test/e2e/install/install_test.go

```diff
@@ -90,12 +92,25 @@ var _ = Describe("[Install] continuousIntegration", func() {
 			}, 5*time.Minute, 10*time.Second).Should(BeTrue(), "RTE condition did not become available")
 
 			By("checking the NRT CRD is deployed")
-			crd := &apiextensionv1.CustomResourceDefinition{}
-			key := client.ObjectKey{
-				Name: crdName,
-			}
-			err := e2eclient.Client.Get(context.TODO(), key, crd)
+			_, err := crds.GetByName(e2eclient.Client, crdNRTName)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("checking the NRO CRD is deployed")
+			_, err = crds.GetByName(e2eclient.Client, crdNROName)
 			Expect(err).NotTo(HaveOccurred())
+
+			By("checking Daemonset is up&running")
+			workerNodes, err := getWorkerNodes()
+			Expect(err).NotTo(HaveOccurred())
+
+			ds, err := getDaemonSetByOwnerReference(updatedNROObj.UID)
+			Expect(err).NotTo(HaveOccurred())
+
+			workerNodesNumber := int32(len(workerNodes))
```

Comment:
rather than using this value, we should use the `desiredNumberScheduled` as target. We need to assume the value is correctly computed (and maybe test it on a separate test). IOW, we don't need to count the worker nodes (which we are doing incorrectly here).

Reply:
Ok, understood. I will change it.

For the sake of curiosity... will this be a proper way to found the nodes where daemonset should be running ?

```go
func getWorkerNodes(labelSelector map[string]string) ([]corev1.Node, error) {
	nodeList := &corev1.NodeList{}

	if err := e2eclient.Client.List(context.TODO(), nodeList, &client.ListOptions{LabelSelector: labels.SelectorFromSet(labelSelector)}); err != nil {
		return nil, fmt.Errorf("failed to get nodes: %w", err)
	}
	return nodeList.Items, nil
}
```
using  the MachineConfigPool.Spec.NodeSelector.MatchLabels as `labelSelector`

Reply:
there is code in the `controllers` package (soon to be moved on its own package) to resolve nodegroups which we can use here